### PR TITLE
Localflavor Nepal: Support for nepal added in django-localflavor

### PR DESCRIPTION
**Pull request by vishalpandeyvip**
**Support for Nepal Added as requested by this issue #414  **

What Fields are added ?

- [x] Field for zones of Nepal.
- [x] Field for provinces of Nepal.
- [x] Field for districts of Nepal.
- [x] Field for Postal Code of Nepal.

**All Changes**

- [x] Add an entry to the docs/changelog.rst describing the change.

- [x] Add an entry for your name in the docs/authors.rst file if it's not
      already there.

**New Fields Only**

- [x] Prefix the country code to all fields.

- [x] Field names should be easily understood by developers from the target
      localflavor country. This means that English translations are usually
      not the best name unless it's for something standard like postal code,
      tax / VAT ID etc.

- [x] Prefer '<country code>PostalCodeField' for postal codes as it's
      international English; ZipCode is a term specific to the United
      States postal system.

- [x] Add meaningful tests. 100% test coverage is not required but all
      validation edge cases should be covered.

- [x] Add `.. versionadded:: <next-version>` comment markers to new
      localflavors.

- [x] Add documentation for all fields.

### CODE REVIEW

**Comment by benkonrath** [comment]
I'll be a able to take a look at this next week. Thanks for the PR!

File: localflavor/np/models.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+class NPDistrictField(models.CharField):
+    """
+        A Modelfield that provides an option to select 
+        a district from the list of all Nepali districts.
+        .. versionadded:: 4.0
+    """
+
+    def __init__(self, *args, **kwargs):
+        kwargs['choices'] = DISTRICTS
+        kwargs['max_length'] = 255
```
Comment (benkonrath) [inline localflavor/np/models.py:51]: The `max_length` should be set to the max value in the `DISTRICTS` choice. This is for all of the `max_length`s - it should be set to an appropriate value for the data being used.

File: localflavor/np/np_districts.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+Choices for districts are in this format:
+
+    (('name_of_district'), _('Name of district')),
+"""
+
+from django.utils.translation import gettext_lazy as _
+
+# list of districts in Bagmati Zone
+BAGMATI_DISTRICTS = [ 
+    (('bhaktapur'), _('Bhaktapur')),
```
Comment (benkonrath) [inline localflavor/np/np_districts.py:15]: The extra set of brackets aren't needed. It should look like this:
```suggestion
    ('bhaktapur', _('Bhaktapur')),
```
This applies to other places as well.

File: localflavor/np/models.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        self.validators.append(NPPostalCodeValidator())
+
+    def formfield(self, **kwargs):
+        defaults = {'form_class': NPPostalCodeFormField}
+        defaults.update(kwargs)
+        return super().formfield(**defaults)
+
+class NPDistrictField(models.CharField):
+    """
+        A Modelfield that provides an option to select
```
Comment (benkonrath) [inline localflavor/np/models.py:44]:
```suggestion
        A model field that provides an option to select 
```
The same suggestion in other places as well.
